podman_compose.py

```diff
@@ -2219,6 +2219,13 @@ def _parse_compose_file(self) -> None:
             log.debug(" ** merged:\n%s", json.dumps(compose, indent=2))
         # ver = compose.get('version')
 
+        self.x_podman = compose.get("x-podman", {})
+        self.x_podman.update({
+            key.removeprefix("PODMAN_COMPOSE_").lower(): value  # type: ignore[misc]
```

Comment:
I think it makes sense to have an explicit list of allowed environment variables and throw a warning if unsupported one is passed. This solves the problem of people making types and wasting lots of time later.

Also, this commit needs a newsfragment.